### PR TITLE
book commit

### PR DESCRIPTION
**Pull request by DenisBogusevich**

### CODE REVIEW

**Review by RomanYuzviak — APPROVED** [review]
Looks great!

File: pom.xml [hunk 1]
```diff
@@ -72,11 +97,7 @@
...
             </plugin>
-            <plugin>
-                <groupId>org.springframework.boot</groupId>
-                <artifactId>spring-boot-maven-plugin</artifactId>
-            </plugin>
         </plugins>
     </build>
 
-</project>
+</project>
```
Comment (liliia-ponomarenko) [inline pom.xml]: add one empty line

File: src/main/java/onlinebookstore/exception/DataProcessingException.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package onlinebookstore.exception;
+
+public class DataProcessingException extends RuntimeException {
+    public DataProcessingException(String s, Exception e) {
```
Comment (liliia-ponomarenko) [inline src/main/java/onlinebookstore/exception/DataProcessingException.java]: Don’t  use abbreviations 
```suggestion
    public DataProcessingException(String message, Exception exception) {
```

File: src/main/java/onlinebookstore/repository/impl/BookRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
+package onlinebookstore.repository.impl;
+
+import jakarta.persistence.EntityNotFoundException;
```
Comment (liliia-ponomarenko) [inline src/main/java/onlinebookstore/repository/impl/BookRepositoryImpl.java]: better create your custom EntityNotFoundException

File: src/main/java/onlinebookstore/service/impl/BookServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package onlinebookstore.service.impl;
+
+import java.util.List;
+import onlinebookstore.entity.Book;
+import onlinebookstore.repository.BookRepository;
+import onlinebookstore.service.BookService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
```
Comment (liliia-ponomarenko) [inline src/main/java/onlinebookstore/service/impl/BookServiceImpl.java]:
```suggestion
@Service
```
Use annotation Service for classes  with business logic

File: src/main/java/onlinebookstore/service/impl/BookServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+
+    @Autowired
```
Comment (liliia-ponomarenko) [inline src/main/java/onlinebookstore/service/impl/BookServiceImpl.java]: As of Spring Framework 4.3, an @Autowired annotation on such a constructor is no longer necessary if the target bean defines only one constructor to begin with. See [note here ](https://docs.spring.io/spring-framework/reference/core/beans/annotation-config/autowired.html)Let’s check it ;)

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s fix a few mistakes ;)

**Review by liliia-ponomarenko — APPROVED** [review]
LGTM 🔥

**Pull request closed without merging**